### PR TITLE
Add Serial tests for CNCC on OpenStack

### PR DESCRIPTION
**Pull request by andreaskaris**
Signed-off-by: Andreas Karis <ak.karis@gmail.com>

### CODE REVIEW

File: ci-operator/config/openshift/cloud-network-config-controller/openshift-cloud-network-config-controller-master.yaml [hunk 1]
```diff
@@ -66,6 +66,28 @@ tests:
...
+      TEST_SUITE: openshift/conformance/serial
+    workflow: openshift-e2e-openstack-ipi
+- as: e2e-openstack-ovn-serial
+  optional: true
+  steps:
+    cluster_profile: openstack-vexxhost
+    env:
+      BASE_DOMAIN: shiftstack.devcluster.openshift.com
+      CONFIG_TYPE: minimal
+      NETWORK_TYPE: OVNKubernetes
```
Comment (jluhrsen) [inline ci-operator/config/openshift/cloud-network-config-controller/openshift-cloud-network-config-controller-master.yaml:87]: don't think you need this for 4.12+ OVNK should be there by default

**Comment by andreaskaris** [comment]
There's a problem with those rehearsal jobs:

a) when I run the full serial test suite against OSP, about 80 or more tests fail.

b) I can run the following manually:
~~~
./openshift-tests run openshift/conformance/serial --run "\[sig-network\]\[Feature:EgressIP\]" --provider '{"type":"openstack"}'
~~~

But when I add `TEST_ARGS: --run "\[sig-network\]\[Feature:EgressIP\]"` to the arguments of the job then the suite quotes the parameter and that breaks things:
~~~
 + openshift-tests run openshift/conformance/serial --run '"\[sig-network\]\[Feature:EgressIP\]"' --provider '{"type":"openstack"}' -o /logs/artifacts/e2e.log --junit-dir /logs/artifacts/junit
warning: KUBE_TEST_REPO_LIST may not be set when using openshift-tests and will be ignored
openshift-tests version: v4.1.0-5662-g0e21e34
Suite run returned error: suite "openshift/conformance/serial" does not contain any tests
error: suite "openshift/conformance/serial" does not contain any tests 
~~~

**Comment by andreaskaris** [comment]
The tests now fail with:
~~~
 INFO[2022-08-24T17:29:56Z] Failed to discover available identity versions when contacting https://identity.rdo.mtl2.vexxhost.net/v3. Attempting to parse version from URL.
Gateway Timeout (HTTP 504) 
~~~

I think that's a problem with access to vexxhost itself ...

**Comment by andreaskaris** [comment]
Well all the OSP tests seem to be affected by that, not jus the rehearse jobs:
https://search.ci.openshift.org/?search=failed+to+discover+available+identity+versions+when+contacting&maxAge=48h&context=1&type=bug%2Bissue%2Bjunit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

**Comment by jcaamano** [comment]
I have seen this test flaking: `pods should have the assigned EgressIPs and EgressIPs can be deleted and recreated`
This flake sometimes but seldom make the jobs fail.

The 4.12 and 4.13 jobs have not passed once, but mostly due to invariants or infra issues that have little to do with egress ip tests 
afaik.

The master sdn and ovn jobs have passed at least once, and again when they have not it has been mostly due to same reasons as above.

Overall the jobs same defined well to me.

/lgtm

**Comment by jcaamano** [comment]
@jluhrsen 

PTAL

File: ci-operator/step-registry/openshift/e2e/openstack/ipi-e2e-only/OWNERS [hunk 2]
```diff
@@ -0,0 +1 @@
+../OWNERS
```
Comment (pierreprinetti) [inline ci-operator/step-registry/openshift/e2e/openstack/ipi-e2e-only/OWNERS:1]: You should replace this symlink with an actual OWNERS file with your approvers and your reviewers

Comment (pierreprinetti) [inline ci-operator/step-registry/openshift/e2e/openstack/ipi-e2e-only/OWNERS:1]: Replace this symlink with an actual OWNERS file containing your approvers and your reviewers

**Review by pierreprinetti — CHANGES_REQUESTED** [review]
Creating and maintaining your own workflow is one valid option.

Have you considered these alternatives?

### Alternative 1: override the `test` stanza in config
Use the existing `openshift-e2e-openstack-ipi` workflow, [override](https://docs.ci.openshift.org/docs/architecture/step-registry/#ci-operator-test-configuration) the `test` part of it:
```yaml
- as: e2e-openstack-sdn-serial
  optional: true
  steps:
    cluster_profile: openstack-vexxhost
    env:
      BASE_DOMAIN: shiftstack.devcluster.openshift.com
      CONFIG_TYPE: minimal
      NETWORK_TYPE: OpenShiftSDN
      TEST_ARGS: --run \[sig-network\]\[Feature:EgressIP\]
      TEST_SUITE: openshift/conformance/serial
    workflow: openshift-e2e-openstack-ipi
    test:
    - ref: openshift-e2e-test # This overrides the test chain defined in openshift-e2e-openstack-ipi
```

### Alternative 2: skip all openstack tests
Turn `openstack-test-openstack` into a no-operation by setting `OPENSTACK_TEST_SKIPS` to `.*` in config (a slight modification of `openstack-test-openstack` is needed not to make it error).
```yaml
- as: e2e-openstack-sdn-serial
  optional: true
  steps:
    cluster_profile: openstack-vexxhost
    env:
      BASE_DOMAIN: shiftstack.devcluster.openshift.com
      CONFIG_TYPE: minimal
      NETWORK_TYPE: OpenShiftSDN
      TEST_ARGS: --run \[sig-network\]\[Feature:EgressIP\]
      TEST_SUITE: openshift/conformance/serial
      OPENSTACK_TEST_SKIPS: .*
    workflow: openshift-e2e-openstack-ipi
```

---

The only change request is for the OWNERS file, that you will likely want to be able to maintain without asking reviews to our team.
Once that is fixed (and you optionally have considered the alternatives above), this LGTM as-is.

**Comment by jcaamano** [comment]
Following through with the rework on behalf of @andreaskaris here: https://github.com/openshift/release/pull/31724

**Pull request closed without merging**